assets/js/components/dashboard-sharing/DashboardSharingDialog/index.js

```diff
@@ -116,27 +132,61 @@ export default function DashboardSharingDialog() {
 	const closeResetDialog = useCallback( () => {
 		setValue( RESET_SETTINGS_DIALOG, false );
 		openSettingsDialog();
+		setShouldFocusResetButton( true );
 	}, [ openSettingsDialog, setValue ] );
 
 	const closeDialog = useCallback( () => {
 		if ( resetDialogOpen ) {
 			closeResetDialog();
-
 			return null;
 		}
 
 		closeSettingsDialog();
 	}, [ closeResetDialog, closeSettingsDialog, resetDialogOpen ] );
 
+	// Handle escape key for reset dialog
+	useKey(
+		( event ) => resetDialogOpen && ESCAPE === event.keyCode,
+		() => {
+			closeResetDialog();
+		}
+	);
+
+	// Handle clicking on the scrim (outside the dialog)
+	useEffect( () => {
+		const handleScrimClick = ( event ) => {
+			if (
+				resetDialogOpen &&
+				event.target.classList.contains( 'mdc-dialog__scrim' )
+			) {
+				closeResetDialog();
+			}
+		};
+
+		document.addEventListener( 'click', handleScrimClick );
+
+		return () => {
+			document.removeEventListener( 'click', handleScrimClick );
+		};
+	}, [ resetDialogOpen, closeResetDialog ] );
+
 	return (
 		<Portal>
 			<Dialog
 				open={ settingsDialogOpen || resetDialogOpen }
 				onClose={ closeDialog }
 				className="googlesitekit-dialog googlesitekit-sharing-settings-dialog"
 				style={ dialogStyles }
+				// Prevent default modal behavior as we are simulating multiple modals within a single modal here for the settings and reset dialogs.
```

Comment:
Use `{ /* ... */ }` for jsx comments.

Reply:
I can't use `{ }` here because it's within the component prop list, not within the component structure, but I have updated it to `/* */`